### PR TITLE
General updates

### PR DESCRIPTION
**Pull request by alanking**
- [x] Investigate publicly reachable address issue with ZMT
- [x] Authenticate the user in the iCommands container automatically
- [x] Remove "old" directory
---
Some things to consider...
- ZMT must specify an externally reachable address for the nginx reverse-proxy in order to work for some reason
- I used a submodule for ZMT :) Also, ZMT needs updates to work with the recently released 0.9.2 REST C++ client
- iRODS catalog uses postgres:12; seems to work, but I've heard concerns expressed over this
- iCommands have a default irods_environment.json, but does NOT authenticate by default, requiring the user to enter a password or `iinit`. This can be done by passing `-it` when invoking commands through `docker exec`
- iRODS catalog and Metalnx database could reside in the same postgres instance, but for now they are separate services
- All existing files have been moved under a new directory called `old`, but we have the history rewrite to remove binaries/packages to consider

### CODE REVIEW

**Comment by trel** [comment]
- ZMT externally reachable address seems... weird
- submodule okay until we get a ZMT image out there...
- no issues with iRODS 4.3.0 and postgres:12.
- why not pre-initialize `echo rods | iinit`?
- separate databases are fine... metalnx will lose its own soon enough
- why keep things in 'old' at all?  we're in git...

**Comment by alanking** [comment]
Added a TODO list to the OP...

**Comment by alanking** [comment]
Still can't get ZMT to cooperate unless I use the public hostname for my work machine in the `REACT_APP_REST_API_URL`. I wonder if this issue exists on a browser running on the same machine as the containers? I'm running the containers on my work machine and viewing through a browser on my laptop.

**Comment by trel** [comment]
I think that... makes sense?    It is rewriting its own URLs based on that setting, and it has to be renderable / viewable by someone 'out there'...  I think it's working as intended?

**Comment by trel** [comment]
Aka, a 'real' install would always have a real address.

**Comment by alanking** [comment]
I think so, yes.

As I am not near my workstation, it is hard to tell if the demo works "hands free" by using the existing value of `nginx-reverse-proxy` or `${hostname}` or even `localhost` instead, while browsing to `http://localhost:9000` to view ZMT. Perhaps I can try it out tomorrow.

File: metalnx/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+RUN apt-get update && \
+    apt-get install -y \
+        netcat \
+        postgresql-client \
+    && \
+    apt-get clean
+
+COPY metalnx_entrypoint.sh /
+RUN chmod +x /metalnx_entrypoint.sh
+CMD ["/metalnx_entrypoint.sh"]
```
Comment (alanking) [inline metalnx/Dockerfile:12]: Consider adding a comment explaining that this overrides the `CMD` specified in the base image (which does not specify an `ENTRYPOINT`).

File: irods_client_icommands/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/*
+
+RUN mkdir -p /root/.irods
+COPY irods_environment.json /root/.irods
+
+WORKDIR /
+COPY entrypoint.sh .
+RUN chmod +x ./entrypoint.sh
```
Comment (alanking) [inline irods_client_icommands/Dockerfile:34]: `chmod +x` -> `chmod u+x`

File: irods_catalog_provider/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        irods-runtime=${irods_package_version} \
+        irods-server=${irods_package_version} \
+    && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/*
+
+COPY setup-${irods_version}.input /
+RUN mv /setup-${irods_version}.input /irods_setup.input
+
+COPY entrypoint.sh /
```
Comment (alanking) [inline irods_catalog_provider/Dockerfile:48]: Consider using `WORKDIR /` here and relative paths afterwards

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
+version: '3'
+
+services:
+    catalog:
+        build:
+            context: catalog
+        ports:
+            - "5430:5432"
```
Comment (alanking) [inline docker-compose.yml:8]: Leave a comment explaining that we have to expose a different port due to the `metalnx-db` service. Eventually that service will go away and with it we can remove this stanza.

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        ports:
+            - "80:80"
+        depends_on:
+            - irods-client-rest-cpp
+
+    irods-client-zmt:
+        build:
+            context: irods_client_zone_management_tool
+        environment:
+            - REACT_APP_REST_API_URL=http://nginx-reverse-proxy/irods-rest/0.9.2
```
Comment (alanking) [inline docker-compose.yml:46]: TODO: Need to test to make sure this works when browsing using the host running running the containers. If it is not, I'm guessing `${hostname}` or `localhost` will work, but, again, only if the browser is on the same host as the containers.

File: irods_client_rest_cpp/irods_client_rest_cpp.Dockerfile [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+RUN apt-get update && \
+    apt-get install -y irods-client-rest-cpp=${package_version} && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/*
+
+# Use the default rsyslog configuration for this image
+RUN cp /etc/irods_client_rest_cpp/irods_client_rest_cpp.conf.rsyslog /etc/rsyslog.d/00-irods_client_rest_cpp.conf && \
+    cp /etc/irods_client_rest_cpp/irods_client_rest_cpp.logrotate /etc/logrotate.d/irods_client_rest_cpp
+
+COPY entrypoint.sh /
```
Comment (alanking) [inline irods_client_rest_cpp/irods_client_rest_cpp.Dockerfile:35]: Consider using `WORKDIR /` here and relative paths afterwards

File: metalnx/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,12 @@
+FROM irods/metalnx:2.6.0
+
+RUN apt-get update && \
+    apt-get install -y \
+        netcat \
+        postgresql-client \
+    && \
+    apt-get clean
+
+COPY metalnx_entrypoint.sh /
```
Comment (alanking) [inline metalnx/Dockerfile:10]: Consider using `WORKDIR /` here and relative paths afterwards

**Comment by alanking** [comment]
All that is left is to confirm whether ZMT behaves using the existing values. I renamed the `catalog` service to `irods-catalog` for clarity. Marking as ready for review.

File: docker-compose.yml (hunk 5 above)
Comment (alanking) [reply docker-compose.yml]: Confirmed

File: .gitmodules [hunk 8]
```diff
@@ -0,0 +1,3 @@
+[submodule "irods_client_zone_management_tool"]
+	path = irods_client_zone_management_tool
+	url = https://github.com/irods/irods_client_zone_management_tool
```
[the remark this answers is not in the document]
Comment (alanking) [reply .gitmodules:3]: Okay. Would a line like this in the README suffice?

> Run the following before attempting to start the Compose project:
> ```
> git submodule update --init
> ```

File: irods_client_icommands/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && \
+    apt-get install -y \
+        apt-transport-https \
+        gnupg \
+        wget \
+  && \
+  apt-get clean && \
+  rm -rf /var/lib/apt/lists/* /tmp/*
```
[the remark this answers is not in the document]
Comment (alanking) [reply irods_client_icommands/Dockerfile:12]: Will fix

File: irods_client_icommands/entrypoint.sh [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+do
+    sleep 1
+done
+
+sleep 3
+
+until nc -z ${irods_catalog_provider_hostname} 1247
+do
+    sleep 1
+done
```
[the remark this answers is not in the document]
Comment (alanking) [reply irods_client_icommands/entrypoint.sh:14]: Will do

File: .gitmodules (hunk 8 above)
[the remark this answers is not in the document]
Comment (korydraughn) [reply .gitmodules:3]: Yep. Just a line that helps them not fall in the git submodule abyss.

And a comment stating how often it should be run? (e.g. should they run that one time after cloning the project or before running docker-compose?)

[the remark this answers is not in the document]
Comment (alanking) [reply .gitmodules:3]: I've included a blurb in the README which also mentions when `git submodule update` should be run. Let's move the discussion there if there are any changes to be had.

**Review by korydraughn — APPROVED** [review]
Squash and pound when ready.

**Pull request merged**